src/plugins/dotplot/src/DotPlotWidget.cpp

```diff
@@ -778,7 +779,16 @@ bool DotPlotWidget::sl_showSettingsDialog(bool disableLoad) {
 
 // ask user if he wants to save dotplot first
 void DotPlotWidget::sl_showDeleteDialog() {
-    int answer = QMessageBox::information(this, tr("Save dot-plot"), tr("Save dot-plot data before closing?"), QMessageBox::Yes, QMessageBox::No, QMessageBox::Cancel);
+    if (deleteDialogWasShownAndNotCancelled) {
```

Comment:
Can you check this on the caller side. 
The current logic has a logical flow: a caller calls 'showDeleteDialog' and nothing happens because of some internal magic.
A better solution would be:
1. Internal logic asserts (SAFE_POINT) when dialog can't be shown
2. The caller checks if the dialog can be shown or not and shows it only if it can.